File: 08/main.py
with open("input.txt") as f:
    data = [tuple([[y for y in x.split(" ") if y]
            for x in l.split("|")])
            for l in f.read().split("\n")]

out = 0
for l in data:
    for x in l[1]:
        if len(x) in [2, 3, 4, 7]:
            out += 1

print(out)

File: 08/main.cpp
#include "../common.h"
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
    ifstream fp("input.txt");
    vector<string> lines = read_lines(fp);

    int out = 0;

    for(string line : lines) {
        line.erase(0, line.find(" | ")+3);

        for(string word : split(line, " ")) {
            int l = word.length();
            if((l==2) || (l==3) || (l==4) || (l==7)) {
                out++;
            }
        }
    }
    cout << out << "\n";
}